Modules/Loadable/Volumes/qSlicerVolumesReader.cxx

```diff
@@ -186,7 +191,14 @@ bool qSlicerVolumesReader::load(const IOProperties& properties)
         }
       if (appLogic)
         {
-        appLogic->PropagateVolumeSelection(); // includes FitSliceToAll by default
+        if (fit == 1)
+          {
+          appLogic->PropagateVolumeSelection(); // includes FitSliceToAll by default
+          }
+        else if (fit == 2)
+          {
+          appLogic->FitSlicesToVolume(node->GetID());
```

Comment:
If you choose fit==2 (by the way, I agree with Steve, these should be enums instead of magic numbers) then what would select the new volume as foreground/background/label in any of the slice views?